### PR TITLE
Update compiling_for_android.rst

### PR DESCRIPTION
**Pull request by AndyAnon14**
Updated 'compiling_for_adroid.rst' to include instructions on how to compile the Godot Editor for Android.

_Bugsquad edit: salvages https://github.com/godotengine/godot-docs/pull/6089_

### CODE REVIEW

File: development/compiling/compiling_for_android.rst [hunk 1]
```diff
@@ -206,6 +206,53 @@ file in the ``bin\`` directory of your Godot source folder, so that the
 next time you build you will automatically have the custom templates
 referenced.
 
+Building the Godot Editor
+-------------------------
```
Comment (Calinou) [inline development/compiling/compiling_for_android.rst:210]:
```suggestion
Building the Godot editor
-------------------------
```

File: development/compiling/compiling_for_android.rst [hunk 2]
```diff
@@ -206,6 +206,53 @@ file in the ``bin\`` directory of your Godot source folder, so that the
...
+
+Compiling the editor is done by calling SCons from the Godot
+root directory with the following arguments:
+
+-  Release Debug target (the editor will **only** compile with the **release debug** target.)
```
Comment (Calinou) [inline development/compiling/compiling_for_android.rst:215]: I'd remove this notice as `target=debug` can also be used.

File: development/compiling/compiling_for_android.rst [hunk 3]
```diff
@@ -206,6 +206,53 @@ file in the ``bin\`` directory of your Godot source folder, so that the
...
+::
+
+    cd platform/android/java
+    # On Windows
+   .\gradlew cleanGodotEditor
+   # On Linux and macOS
+   ./gradlew cleanGodotEditor
+
+Installing the Godot Editor
+-------------------------
```
Comment (Calinou) [inline development/compiling/compiling_for_android.rst:246]:
```suggestion
Installing the Godot editor
---------------------------
```

File: development/compiling/compiling_for_android.rst [hunk 4]
```diff
@@ -206,6 +206,53 @@ file in the ``bin\`` directory of your Godot source folder, so that the
...
+   # On Windows
+   .\gradlew generateGodotEditor
+   # On Linux and macOS
+   ./gradlew generateGodotEditor
+
+
+The resulting APK will be located at ``bin/android_editor.apk``.
+
+Cleaning the Editor templates
+-----------------------------
```
Comment (Calinou) [inline development/compiling/compiling_for_android.rst:233]:
```suggestion
Removing the Editor templates
-----------------------------
```

**Review by mhilbrunner — APPROVED** [review]
Looks good to me (besides Calinou's suggestion above :))

File: development/compiling/compiling_for_android.rst (hunk 2 above)
Comment (AndyAnon14) [reply development/compiling/compiling_for_android.rst]: I removed the notice as requested. You are correct that there are additional targets that the editor can be built on.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you, and congrats on your first contribution to Godot's docs! 🎉

**Pull request merged**